src/cmd/flux-exec.c

```diff
@@ -226,18 +287,28 @@ static void stdin_cb (flux_reactor_t *r,
     struct fbuf *fb = fbuf_read_watcher_get_buffer (w);
     flux_subprocess_t *p;
     const char *ptr;
-    int lenp;
+    int len, lenp;
+    int min_credits = subprocess_min_credits ();
 
-    if (!(ptr = fbuf_read (fb, -1, &lenp)))
+    if (!(ptr = fbuf_read (fb, min_credits, &lenp)))
```

Comment:
I think this is a yes but are we certain `min_credits` is always > 0 here?

Reply:
hmmmmm.  It should be impossible, discounting unknown or future bugs.  Think we should assert check just in case/

Reply:
Maybe nothing is needed, but I suppose it could just stop the watcher and return if credits is zero.

Reply:
that sounds like a decent idea.

Reply:
oh wait!  i think it is possible to be 0, although extremely unlikely, when the stream is closed.  should add some smarts in there for that case.

Reply:
ok, i had to think about it a bit.  I think it is possible min_credits could be zero if the remote subprocess has a fatal error (like libsubprocess ENOMEM kinda thing), so we're at 0 credits locally and never get credits back.  But since this is only in an error scenario, I don't think it matters.

Although, this did make me think of a corner case to fix.  If I call `subprocess_remove_credits()`, it's possible that `min_credits` may no longer be zero.  So I should re-check and start the `stdin_w` if necessary.